Mathlib/Probability/Moments/Tilted.lean

```diff
@@ -0,0 +1,153 @@
+
+* `integral_tilted_mul_self`: the integral of `X` against the tilted measure `μ.tilted (t * X ·)`
+  is the first derivative of the cumulant generating function of `X` at `t`.
+  `(μ.tilted (t * X ·))[X] = deriv (cgf X μ) t`
+* `variance_tilted_mul`: the variance of `X` under the tilted measure `μ.tilted (t * X ·)`
+  is the second derivative of the cumulant generating function of `X` at `t`.
+  `Var[X; μ.tilted (t * X ·)] = iteratedDeriv 2 (cgf X μ) t`
+
+-/
+
+
+open MeasureTheory Real Set Finset
+
+open scoped NNReal ENNReal ProbabilityTheory
+
+variable {Ω : Type*} {mΩ : MeasurableSpace Ω} {μ ν : Measure Ω} {X : Ω → ℝ} {t u : ℝ}
+
+namespace ProbabilityTheory
+
+section Apply
+
+/-! ### Apply lemmas for `tilted` expressed with `mgf` or `cgf`. -/
+
+lemma tilted_mul_apply_mgf' {s : Set Ω} (hs : MeasurableSet s) :
+    μ.tilted (t * X ·) s = ∫⁻ a in s, ENNReal.ofReal (exp (t * X a) / mgf X μ t) ∂μ := by
+  rw [tilted_apply' _ _ hs, mgf]
+
+lemma tilted_mul_apply_mgf [SFinite μ] (s : Set Ω) :
+    μ.tilted (t * X ·) s = ∫⁻ a in s, ENNReal.ofReal (exp (t * X a) / mgf X μ t) ∂μ := by
+  rw [tilted_apply, mgf]
+
+lemma tilted_mul_apply_cgf' [NeZero μ] {s : Set Ω} (hs : MeasurableSet s)
+    (ht : Integrable (fun ω ↦ exp (t * X ω)) μ) :
+    μ.tilted (t * X ·) s = ∫⁻ a in s, ENNReal.ofReal (exp (t * X a - cgf X μ t)) ∂μ := by
+  simp_rw [tilted_mul_apply_mgf' hs, exp_sub, exp_cgf ht]
+
+lemma tilted_mul_apply_cgf [NeZero μ] [SFinite μ] (s : Set Ω)
+    (ht : Integrable (fun ω ↦ exp (t * X ω)) μ) :
+    μ.tilted (t * X ·) s = ∫⁻ a in s, ENNReal.ofReal (exp (t * X a - cgf X μ t)) ∂μ := by
+  simp_rw [tilted_mul_apply_mgf s, exp_sub, exp_cgf ht]
+
+lemma tilted_mul_apply_eq_ofReal_integral_mgf' {s : Set Ω} (hs : MeasurableSet s) :
+    μ.tilted (t * X ·) s = ENNReal.ofReal (∫ a in s, exp (t * X a) / mgf X μ t ∂μ) := by
+  rw [tilted_apply_eq_ofReal_integral' _ hs, mgf]
+
+lemma tilted_mul_apply_eq_ofReal_integral_mgf [SFinite μ] (s : Set Ω) :
+    μ.tilted (t * X ·) s = ENNReal.ofReal (∫ a in s, exp (t * X a) / mgf X μ t ∂μ) := by
+  rw [tilted_apply_eq_ofReal_integral _ s, mgf]
+
+lemma tilted_mul_apply_eq_ofReal_integral_cgf' [NeZero μ] {s : Set Ω} (hs : MeasurableSet s)
+    (ht : Integrable (fun ω ↦ exp (t * X ω)) μ) :
+    μ.tilted (t * X ·) s = ENNReal.ofReal (∫ a in s, exp (t * X a - cgf X μ t) ∂μ) := by
+  simp_rw [tilted_mul_apply_eq_ofReal_integral_mgf' hs, exp_sub]
+  rwa [exp_cgf]
+
+lemma tilted_mul_apply_eq_ofReal_integral_cgf [NeZero μ] [SFinite μ] (s : Set Ω)
+    (ht : Integrable (fun ω ↦ exp (t * X ω)) μ) :
+    μ.tilted (t * X ·) s = ENNReal.ofReal (∫ a in s, exp (t * X a - cgf X μ t) ∂μ) := by
+  simp_rw [tilted_mul_apply_eq_ofReal_integral_mgf s, exp_sub]
+  rwa [exp_cgf]
+
+end Apply
+
+section Integral
+
+/-! ### Integral of `tilted` expressed with `mgf` or `cgf`. -/
+
+variable {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E]
+
+lemma setIntegral_tilted_mul_mgf' (g : Ω → E) {s : Set Ω} (hs : MeasurableSet s) :
```

Comment:
I don't see how the lemma name corresponds to the statement. Maybe `setIntegral_tilted_mul_eq_mgf'`? (oterhwise, it seems to say you're multiplying the mgf by something). Same thing below.